### DISCUSSION TITLE
how to start netdata as docker within Synology DSM7

### DISCUSSION THREAD

**Discussion by binderth** [netdata/netdata#12237] [Q&A]
I'd like to run netdata within the docker provided with Synology's DSM7.

For that I downloaded, the dockerfile from https://registry.hub.docker.com/r/netdata/netdata/ 
I added the directories like described in the docker script:
![grafik](https://user-images.githubusercontent.com/9037329/155538138-bc0bc46f-8235-4d49-a999-920ce1b23d83.png)

the container won't start and throws this error:
```
Start container netdata failed: {"message":"OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:61: mounting \"/volume1/docker/netdata/proc\" to rootfs at \"/proc\" caused: \"/volume1/@docker/btrfs/subvolumes/d0fdec3fda0adaad625b6888b22a548931a15b13308083be892fd4c1fc1aec00/proc\" cannot be mounted because it is not of type proc: unknown"}.
```

Where can I start to search for how to start the docker-container properly in DSM7?

**Comment by ilyam8** [comment]
Hi, @binderth. Is using that UI required? Did you try [docker run or docker-compose commands](https://learn.netdata.cloud/docs/agent/packaging/docker#create-a-new-netdata-agent-container)?

**Reply by binderth** [reply]
As far as I can see, there's no way to use `docker run` for that. There's a tab for arguments to be handed over to the docker container, but unfortunatelty, the container won't start with that. I don't know, what I'm missing.
looks like this for volumes, there's one for ports and environment variables also
![grafik](https://user-images.githubusercontent.com/9037329/155942195-226ade4e-fbee-452f-925c-258a97a28869.png)


But I contacted Synology support, if there's a way to run or import docker scripts/commands into DSM7 docker.

**Reply by ilyam8** [reply]
> But I contacted Synology support

That is the right decision, they will help you. I can't really suggest something because I've never used Synology DSM. Also, Synology DSM [has no /etc/os-release file](https://github.com/netdata/netdata/issues/12068).

---

@timrettop @abl I know you guys are Synology DSM users. Can you help us answer @binderth's question? Did you try to install Netdata as a Docker container?

**Comment by timrettop** [comment]
@ilyam8 sorry I'm running netdata natively, not in docker. 

@binderth have you tried changing the mounts for /proc /sys /etc/passwd and /etc/group to the DSM defaults? (I.e. don't include docker/netdata) and run as privileged?  (Also set read-only on all of them)

As @ilyam8 indicated, you'll need to keep the mount point docker/netdata/etc/os-release the way you have it (but with read-only) and before you start it, you'll need to create the file and set its permissions so its readable by the docker user.  

The contents of the file can be this: 
```NAME="Synology"
VERSION="DSM 7"
ID=synology
ID_LIKE=synology
VERSION_ID="7.0"
```
